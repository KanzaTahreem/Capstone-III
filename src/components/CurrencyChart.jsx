import React from 'react';
import { useSelector } from 'react-redux';
import { Chart } from 'react-charts';

const CurrencyChart = () => {
  const chartData = useSelector((state) => state.chart.chartArray);

  const data = React.useMemo(
    () => {
      const mappedPrices = chartData.prices?.map((price, index) => [
        index === chartData.prices.length - 1 ? 'Today' : `${chartData.prices.length - index}d`,
        price[1],
      ]);
      return [
        {
          label: 'Series 1',
          data: mappedPrices || [],
        },
      ];
    },
    [chartData],
  );

  const axes = React.useMemo(
    () => [
      { primary: true, type: 'ordinal', position: 'bottom' },
      { type: 'linear', position: 'left' },
    ],
    [],
  );

  return (
    <div
      style={{
        height: '400px',
        width: '100%',
      }}
    >
      { chartData.prices && chartData.prices.length ? <Chart data={data} axes={axes} /> : <div /> }
    </div>
  );
};

export default CurrencyChart;
