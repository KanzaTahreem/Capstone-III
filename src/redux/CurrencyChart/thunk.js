import { createAsyncThunk } from '@reduxjs/toolkit';
import { fetchChart, FETCH_CHART } from './chart';

const fetchChartFromApi = createAsyncThunk(FETCH_CHART, async (coinId, thunkApi) => {
  const response = await fetch(`http://localhost:3005/v4/details/chart/${coinId}`, {
    method: 'GET',
  });

  const responseJSON = await response.json();
  thunkApi.dispatch(fetchChart(responseJSON));
  return responseJSON;
});

export default fetchChartFromApi;
