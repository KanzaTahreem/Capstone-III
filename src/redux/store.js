import { configureStore } from '@reduxjs/toolkit';
import coinsReducer from './AllCoins/coins';
import chartReducer from './CoinChart/chart';

const store = configureStore({
  reducer: {
    crypto: coinsReducer,
    chart: chartReducer,
  },
});

export default store;
