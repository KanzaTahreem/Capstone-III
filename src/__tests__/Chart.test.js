import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import CurrencyChart from '../components/CurrencyChart';

describe('Testing Chart', () => {
  let chart;
  beforeAll(async () => {
    chart = renderer
      .create(
        <Provider store={store}>
          <Router>
            <CurrencyChart />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('should renders correctly', () => {
    expect(chart).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(chart).toMatchSnapshot();
  });
});
