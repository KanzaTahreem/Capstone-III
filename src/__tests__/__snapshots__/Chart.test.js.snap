// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Chart should match the snapshot 1`] = `
<div
  style={
    Object {
      "height": "400px",
      "width": "100%",
    }
  }
>
  <div />
</div>
`;
